MetaMask/metamask-extension
# fix: use PORTFOLIO_VIEW flag to determine chain polling

## **Description**

Updates the token price and detection hooks to only poll across chains when `PORTFOLIO_VIEW` is set.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28504?quickstart=1)

## **Related issues**

## **Manual testing steps**

1. With `PORTFOLIO_VIEW=1`, requests should go to the price api across all chains.
2. Without `PORTFOLIO_VIEW=1`, requests should go to the price api on the current chain.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
